types/src/on_chain_config/execution_config.rs

```diff
@@ -152,6 +152,7 @@ pub enum TransactionDeduperType {
 }
 
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "snake_case")] // cannot use tag = "type" as nested enums cannot work, and bcs doesn't support it
```

Comment:
hmm... as long as compat test passes, I guess..

Reply:
This is new for this release. 

But also,  names are only relevant for release tooling, not onchain